app/models/tour_list.rb

```diff
@@ -0,0 +1,39 @@
+class TourList
+  attr_reader :contents
+
+  def initialize(initial_contents)
```

Comment:
Better practice to do 
```ruby
def initialize(initial_contents = Hash.new(0))
  @contents = initial_contents
```
So you don't have to initialize with `(nil)` in the case of no initial contents.

Reply:
Is needed when dealing with sessions, I don't know why but if I make the changes you've suggested, everything fails.